### PR TITLE
Accounting refactor

### PR DESCRIPTION
**Pull request by czue**

### CODE REVIEW

File: corehq/apps/accounting/models.py [hunk 1]
```diff
@@ -751,78 +805,45 @@ def apply_upgrades_and_downgrades(self, downgraded_privileges=None,
...
 
-            send_html_email_async.delay(
-                email_subject,
-                sub_change_email_address,
-                render_to_string('accounting/subscription_change_email.html', email_context),
-                text_content=render_to_string('accounting/subscription_change_email.txt', email_context),
-            )
 
-        subscription_upgrade_or_downgrade.send_robust(self.domain, domain=self.domain)
+    @staticmethod
```
Comment (dimagimon) [inline corehq/apps/accounting/models.py]: E303 too many blank lines (2)

**Comment by czue** [comment]
worked on this with @proteusvacuum and @nickpell 

cc @dimagi/team-commcare-hq

**Comment by kaapstorm** [comment]
Nice!

**Comment by czue** [comment]
I think this is good to go.

**Comment by kaapstorm** [comment]
Well done, @czue, @proteusvacuum, @nickpell!

File: corehq/apps/accounting/models.py [hunk 2]
```diff
@@ -751,78 +805,44 @@ def apply_upgrades_and_downgrades(self, downgraded_privileges=None,
...
+        if Subscriber.should_send_subscription_notification(old_subscription, new_subscription):
+            send_subscription_change_alert(self.domain, new_subscription, old_subscription, internal_change)
 
-            send_html_email_async.delay(
-                email_subject,
-                sub_change_email_address,
-                render_to_string('accounting/subscription_change_email.html', email_context),
-                text_content=render_to_string('accounting/subscription_change_email.txt', email_context),
-            )
+        subscription_upgrade_or_downgrade.send_robust(None, domain=self.domain)
```
Comment (esoergel) [inline corehq/apps/accounting/models.py]: :heart: 
This new version is so much more readable

**Pull request merged**